htmx.ext.shoelace.js

```diff
@@ -1,41 +1,49 @@
-const slTypes = 'sl-button, sl-checkbox, sl-color-picker, sl-input, sl-radio-group, sl-range, sl-rating, sl-select, sl-switch, sl-textarea'
+const slTypes = "sl-button, sl-checkbox, sl-color-picker, sl-input, sl-radio-group, sl-range, sl-rating, sl-select, sl-switch, sl-textarea";
 
 /* Lightly modified version of the same function in htmx.js */
 function shouldInclude(elt) {
-
 	// sl-rating doesn't have a name attribute exposed through the Shoelace API
-	if (elt.tagName == 'SL-RATING' && elt.getAttribute('name')) {
+	if (elt.tagName === "SL-RATING" && elt.getAttribute("name")) {
 		return true;
 	}
-	if (elt.name === "" || elt.name == null || elt.disabled) {
-		return false;
-	}
-	if (elt.type === "submit") {
+
+	if (elt.name === "" || elt.name == null || elt.disabled || elt.type === "submit") {
 		return false;
 	}
-	if (elt.tagName === 'SL-CHECKBOX' || elt.tagName === 'SL-SWITCH') {
+
+	if (elt.tagName === "SL-CHECKBOX" || elt.tagName === "SL-SWITCH") {
 		return elt.checked;
 	}
+
+	if (elt.tagName === "SL-SELECT" || elt.tagName === "SL-RADIO-GROUP") {
+		return elt.value.length > 0;
+	}
+
 	return true;
 }
 
 htmx.defineExtension('shoelace', {
-	onEvent : function(name, evt) {
-		if ((name === "htmx:configRequest") && (evt.detail.elt.tagName == 'FORM')) {
+	onEvent: function(name, evt) {
+		if ((name === "htmx:configRequest") && (evt.detail.elt.tagName === "FORM")) {
 			evt.detail.elt.querySelectorAll(slTypes).forEach((elt) => {
 				if (shouldInclude(elt)) {
-
-					if (elt.tagName === 'SL-CHECKBOX' || elt.tagName === 'SL-SWITCH') {
-						evt.detail.parameters[elt.name] = elt.checked
-
-					} else if (elt.tagName == 'SL-RATING') {
-						evt.detail.parameters[elt.getAttribute('name')] = elt.value
-
-					} else {
-						evt.detail.parameters[elt.name] = elt.value
+					switch (elt.tagName) {
+						case "SL-CHECKBOX":
+						case "SL-SWITCH":
+							evt.detail.parameters[elt.name] = elt.value ?? elt.checked;
```

Comment:
I think maybe this shouldn't have the `?? elt.checked' part. `value` returns on if checked with no value attribute set on the input element.